java/src/main/java/com/databricks/mlflow/utils/SerializationUtils.java

```diff
@@ -0,0 +1,87 @@
+package com.databricks.mlflow.utils;
+
+import java.io.File;
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+/**
+ * Utilities for serializing and deserializing objects to and from various
+ * persistence formats, such as JSON and YAML
+ */
+public class SerializationUtils {
+    private static final ObjectMapper jsonMapper = new ObjectMapper(new JsonFactory());
+    private static final ObjectMapper yamlMapper = new ObjectMapper(new YAMLFactory());
+
+    /**
+     * Produces a JSON string representation of a Java object
+     *
+     * @return A string in valid JSON format
+     */
+    public static String toJson(Object object) throws JsonProcessingException {
+        return jsonMapper.writeValueAsString(object);
+    }
+
+    /**
+     * Produces a Java object representation of a JSON-formatted string
+     *
+     * @param json A string in valid JSON format
+     */
+    public static <T> T fromJson(String json) throws IOException {
+        return jsonMapper.readValue(json, new TypeReference<T>() {});
+    }
+
+    /**
+     * Produces a Java object representation of a JSON-formatted file
+     *
+     * @param filePath The path to the JSON-formatted file
+     * @param class The class of the Java object that should be produced
+     */
+    public static <T> T parseJsonFromFile(String filePath, Class<T> objectClass)
+        throws IOException {
```

Comment:
hypernit: [continuation indents are at +4 indentation](https://google.github.io/styleguide/javaguide.html#s4.5.2-line-wrapping-indent) (compared to default +2) to distinguish from body